components/Modal/index.js

```diff
@@ -1,11 +1,41 @@
-import PropTypes from 'prop-types';
+import React, { useCallback, useEffect } from 'react';
+import classNames from './modal.module.css';
 
-const Modal = () => {
-  return <div> </div>;
-};
+export default function Modal({ showModal, setShowModal, render }) {
+  const openModal = () => {
+    setShowModal((prev) => !prev);
+  };
+  
+  const keyPress = useCallback(
+    (e) => {
+      if (e.key === 'Escape' && showModal) {
+        setShowModal(false);
+      }
+    },
+    [setShowModal, showModal]
+  );
 
-Modal.propTypes = {
-  name: PropTypes.string,
-};
-
-export default Modal;
+  useEffect(() => {
+    document.addEventListener('keydown', keyPress);
+    return () => document.removeEventListener('keydown', keyPress);
+  }, [keyPress]);
+  return (
+    <>
+      {showModal ? (
+        <div>
+          <div onClick={openModal} className={classNames.backdrop}></div>
```

Comment:
In other file we import css as style object/namespace i.e we do styles.backdrop. Can you please check and make these consistent